### PR TITLE
Fix short-circuiting of scope and add name

### LINKED ISSUE #1999: Flow: Failure for scope to be populated if a component fails to load

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

This ensures the scope is filled in a better fashion. There are still edge cases such as component naming issues.

#### Which issue(s) this PR fixes

Closes #1999

### CODE REVIEW

File: pkg/river/vm/vm.go [hunk 1]
```diff
@@ -506,6 +506,9 @@ func (vm *Evaluator) evaluateExpr(scope *Scope, assoc map[value.Value]ast.Node,
 
 // A Scope exposes a set of variables available to use during evaluation.
 type Scope struct {
+	// Name is the name of this particular scope/block
+	Name string
```
Comment (rfratto) [inline pkg/river/vm/vm.go:510]: Sorry, why do we want to name the scopes?

Comment (mattdurham) [reply pkg/river/vm/vm.go:510]: It's mostly debugging, to figure out which one we are on.

Comment (rfratto) [reply pkg/river/vm/vm.go:510]: I'm not sure about this IMO, every component effectively has the same scope 🤔

Comment (mattdurham) [reply pkg/river/vm/vm.go:510]: Isn't each component a separate scope?

Comment (rfratto) [reply pkg/river/vm/vm.go:510]: No, the scope is the object-ized form of the entire graph, each component gets a copy of it at evaluation time.

Comment (mattdurham) [reply pkg/river/vm/vm.go:510]: That's true, it was helpful when I was trying to dig through the code to see what part of the evaluation I was at but I can walk the stack to see that.

File: pkg/flow/internal/controller/loader.go [hunk 2]
```diff
@@ -268,14 +268,15 @@ func (l *Loader) EvaluateDependencies(parentScope *vm.Scope, c *ComponentNode) {
...
 func (l *Loader) evaluate(parent *vm.Scope, c *ComponentNode) error {
 	ectx := l.cache.BuildContext(parent)
-	if err := c.Evaluate(ectx); err != nil {
-		level.Error(l.log).Log("msg", "failed to evaluate component", "component", c.NodeID(), "err", err)
-		return err
-	}
+	err := c.Evaluate(ectx)
 	// Always update the cache both the arguments and exports, since both might
-	// change when a component gets re-evaluated.
+	// change when a component gets re-evaluated. We also want to cache the arguments and exports in case of an error
```
Comment (rfratto) [inline pkg/flow/internal/controller/loader.go:273]: tiniest nit: end comments in periods please :) I'd prefer to treat comments as if they were public-facing documentation.

**Review by rfratto — APPROVED** [review]
🎉 Nice, LGTM. 

Should we have a test for this?

**Pull request merged**